Improve palette options and usage

TheDaemoness opened the pull request:
BREAKING CHANGES:
- Renames nick-colors to identifier-colors.
- Removes command-prompt (unused).
- Removes options for cmodes, umodes, and snomasks from palette.

TheDaemoness committed:
Use nick colors for channel+nick names in windows

TheDaemoness committed:
Rework the default palette and how it's used

TheDaemoness committed:
Implement style overrides for ids and modes

Breaking changes:
- Renames nick-colors to identifier-colors.
- Removes command-prompt (unused).
- Removes options for cmodes, umodes, and snomasks from palette.

Also tweaks a few defaults.

TheDaemoness committed:
Clean up redundant imports

glguy commented on src/Client/Image/Message.hs:
```diff
@@ -1221,3 +1240,11 @@ drawWindowLine palette w padAmt wl = wrap (drawPrefix wl) (view wlImage wl)
+  | isChanPrefix $ Text.head $ idText name = view palCModes
+  | otherwise = view palUModes 
+  where
+    -- TODO: Don't hardcode this, query ISUPPORT.
```
Could we put this parameter into `MessageRendererParams` and route it down into this operation?

glguy committed:
transpose is baked into my brain as being from Data.List

TheDaemoness committed:
Add channel types to MessageRenderParams

This allows these to not be hard-coded in the future,
depending instead on the value of the CHANTYPES ISUPPORT token.

glguy commented:
I've been using this for a few days now and I'm pretty happy with things. I might add separate channel and nick palettes later, but that shouldn't block progress.
